test/helpers.ts

```diff
@@ -0,0 +1,173 @@
+import { constants, BigNumber, utils } from 'ethers'
+import { chai, send, want } from 'minihat'
+import {describe} from "mocha";
+
+export { snapshot, revert, send, wad, ray, rad, apy, N, BANKYEAR, WAD, RAY, RAD, U256_MAX } from 'minihat'
+
+const debug = require('debug')('rico:test')
+const ramp_members = ['vel', 'rel', 'bel', 'cel']
+
+export async function check_gas (gas, minGas, maxGas) {
+  await want(gas.toNumber()).to.be.at.most(maxGas);
+  if( gas.toNumber() < minGas ) {
+    console.log("gas reduction: previous min=", minGas, " gas used=", gas.toNumber());
+  }
+}
+
+//
+// for move patterns
+// verify takes previous dst value
+export function test2D(
+  name     : string,
+  init     : () => void,
+  fillDst  : (prev, next) => Promise<any>,
+  fill     : (prev, next) => Promise<any>,
+  clear    : (prev, next) => Promise<any>,
+  stay     : (prev) => Promise<any>,
+  one      : Buffer | boolean | BigNumber | number,
```

Comment:
Are you aware of `any` type
You might enjoy being fast and loose with your types every once in a while
Nobody will judge you